ironforgedbot/commands/hiscore/calculator.py

```diff
@@ -1,95 +1,159 @@
+            )
     except requests.exceptions.RequestException as e:
-        raise RuntimeError(f'Encountered an error calling Runescape API: {e}')
+        raise RuntimeError(f"Encountered an error calling Runescape API: {e}")
 
     return resp.json()
 
 
-def _get_skills_score(score_data) -> Dict[str, int]:
-    skill_points = {}
+def _get_skills_info(score_data) -> List[Skill]:
+    output = []
+
+    for skill_data in score_data["skills"]:
+        skill_name = skill_data["name"]
 
-    for skill in score_data["skills"]:
-        if not SKILLS.has_value(skill["name"]):
+        if skill_name.lower() == "overall":
             continue
 
-        skill_constant = SKILLS(skill["name"])
-        skill_level = int(skill["level"])
-        experience = int(skill["xp"])
+        skill = SKILLS.get_item_by_name(skill_name)
 
-        if skill_level < 1:
+        if skill is None:
+            logging.info(f"Skill name '{skill_name}' not found")
             continue
 
-        if skill_constant not in SKILL_POINTS_REGULAR or skill_constant not in SKILL_POINTS_PAST_99:
+        if (
+            skill_name not in SKILL_POINTS_REGULAR
+            or skill_name not in SKILL_POINTS_PAST_99
+        ):
+            logging.info(f"No points defined for skill '{skill_name}'")
             continue
 
+        skill_level = int(skill_data["level"]) if int(skill_data["level"]) > 1 else 1
+        experience = int(skill_data["xp"]) if int(skill_data["xp"]) > 0 else 0
+
         if skill_level < 99:
-            points = int(experience / SKILL_POINTS_REGULAR[skill_constant])
+            points = int(experience / SKILL_POINTS_REGULAR[skill_name])
         else:
-            points = (int(LEVEL_99_EXPERIENCE / SKILL_POINTS_REGULAR[skill_constant]) +
-                      int((experience - LEVEL_99_EXPERIENCE) / SKILL_POINTS_PAST_99[skill_constant]))
+            points = int(LEVEL_99_EXPERIENCE / SKILL_POINTS_REGULAR[skill_name]) + int(
+                (experience - LEVEL_99_EXPERIENCE) / SKILL_POINTS_PAST_99[skill_name]
+            )
+
+        output.append(
+            {
+                **skill,
+                **{
+                    "xp": experience,
+                    "level": skill_level,
+                    "points": points,
+                },
+            }
+        )
+
+    return output
+
+
+def _get_activities_info(
+    score_data,
+) -> Tuple[List[Activity], List[Activity], List[Activity]]:
+    clues = []
+    raids = []
+    bosses = []
 
-        if 0 == points:
-            continue
+    for activity in score_data["activities"]:
+        activity_name = activity["name"]
 
-        skill_points[skill_constant] = points
+        if activity_name in IGNORED_ACTIVITIES.values():
+            continue
 
-    return skill_points
+        if not ACTIVITIES.has_item_name(activity_name):
+            logging.warn(f"Activity '{activity_name}' not known")
+            continue
```

Comment:
I don't think we need it, let's say a new activity got added, w/o updating IGNORED_ACTIVITIES, you'll just have a spam in the log from every request.